src/matlab2tikz.m

```diff
@@ -545,17 +559,19 @@ function addStandalone(m2t, fid, part)
     end
 end
 % ==============================================================================
-function str = generateColorDefinitions(names, specs, colorFormat)
+function str = generateColorDefinitions(colorConfig)
     % output the color definitions to LaTeX
-    str = '';
-    ff  = colorFormat;
+    str   = '';
+    names = colorConfig.extraNames;
+    specs = colorConfig.extraSpecs;
+    ff    = colorConfig.format;
+    
+    % make sure to append with '%' to avoid spacing woes
+    FORMAT = ['\\definecolor{%s}{rgb}{' ff ',' ff ',' ff '}%%\n'];
     if ~isempty(names)
-        m2t.content.colors = sprintf('%%\n%% defining custom colors\n');
         colorDef = cell(1, length(names));
         for k = 1:length(names)
-            % make sure to append with '%' to avoid spacing woes
-            colorDef{k}= sprintf(['\\definecolor{%s}{rgb}{', ff, ',', ff, ',', ff,'}%%\n'], ...
-                                  names{k}, specs{k});
+            colorDef{k}= sprintf(FORMAT, names{k}, specs{k});
```

Comment:
I agree (it can't be slower than what we had before, so I don't care). As I remember, the performance-critical stuff is rather in `makeTable` (when working with large datasets at least).